protocol/rpcconsumer/rpcconsumer.go

```diff
@@ -247,10 +248,25 @@ rpcconsumer 127.0.0.1:3333 COS3 tendermintrpc 127.0.0.1:3334 COS3 rest <flags>`,
 			}
 			// handle flags, pass necessary fields
 			ctx := context.Background()
-			networkChainId, err := cmd.Flags().GetString(flags.FlagChainID)
+			var networkChainId string
+			cmdChainIdFlag := viper.GetString(flags.FlagChainID)
+			clientTomlConfig, err := config.ReadFromClientConfig(clientCtx)
 			if err != nil {
 				return err
 			}
+			if cmdChainIdFlag != app.Name {
+				networkChainId = cmdChainIdFlag
+			} else {
+				if clientTomlConfig.ChainID != "" {
+					networkChainId = clientTomlConfig.ChainID
+				} else {
+					networkChainId = app.Name
+				}
+			}
+			if networkChainId != app.Name {
+				utils.LavaFormatInfo("ChainID is different from " + app.Name)
+			}
+
 			logLevel, err := cmd.Flags().GetString(flags.FlagLogLevel)
 			if err != nil {
```

Comment:
Please apply all changes to rpc provider as well.

Reply:
fixed: [17dc045](https://github.com/lavanet/lava/pull/478/commits/17dc0459de269a9e434c5f3411692966a7e9525a)